libs/components/ag-grid/src/lib/modules/ag-grid/ag-grid-wrapper.component.ts

```diff
@@ -287,34 +287,41 @@ export class SkyAgGridWrapperComponent
     }
   }
 
-  #updateGridTheme(themeSettings?: SkyThemeSettings): void {
-    let agTheme: 'default' | 'modern-light' | 'modern-dark';
-    if (themeSettings?.theme.name === 'modern') {
-      agTheme = `modern-${themeSettings.mode.name}` as
-        | 'modern-light'
-        | 'modern-dark';
-    } else {
-      agTheme = `default`;
-    }
-    const variation = this.#hasEditableClass ? 'data-entry-grid' : 'data-grid';
+  #updateGridTheme(
+    hasEditableClass: boolean,
+    themeSettings?: SkyThemeSettings,
+  ): void {
+    const agTheme = agGridTheme(hasEditableClass, themeSettings, this.compact);
+    const previousValue = this.#wrapperClasses.getValue();
+    const previousTheme = previousValue.find((c) => c.startsWith('ag-theme-'));
     let value = [
-      ...this.#wrapperClasses
-        .getValue()
-        .filter((c) => !c.startsWith('ag-theme-')),
-      `ag-theme-sky-${variation}-${agTheme}`,
+      ...previousValue.filter((c) => !c.startsWith('ag-theme-')),
+      agTheme,
     ];
     const textSelectionClass = 'sky-ag-grid-text-selection';
-    if (this.#getTextSelection()) {
+    if (this.#getTextSelection(hasEditableClass)) {
       value.push(textSelectionClass);
     } else {
       value = value.filter((c) => c !== textSelectionClass);
     }
     this.#wrapperClasses.next([...new Set(value)]);
+    if (this.agGrid?.api && !this.agGrid.api.isDestroyed()) {
+      if (this.agGrid?.api?.getGridOption('domLayout') !== 'autoHeight') {
```

Comment:
I thiiiiink typescript is smart enough that you don't need all the `?` before api, since the first if already guarantees it exists

also, why do we care if the `domLayout` is `autoHieght` here? I thought `autoHeight` set the height of the grid to the number of rows x the height of the rows. If the theme changed to compact wouldn't we still want to reset the heights then?

Reply:
The extra `?`'s are due to insufficient mocks in tests. Added a comment for the `autoHeight` bit.

Reply:
interesting about the warning. If we're redrawing all the rows every time do we need to call `resetRowHeights` at all?

Reply:
Right! We don't need to force redraw rows. Good call.